src/inject.js

```diff
@@ -20,11 +20,22 @@ import qs from 'query-string';
 
 console.log('This inject.js has been injected by the shell.');
 
-function initProvider () {
-  const match = window.location.pathname.match(/(0x)?[a-f0-9]{64}|v1/i); // Find the appId
-  const query = qs.parse(window.location.search);
+function getAppId () {
+  // Dapps built into the shell; URL: file://path-to-shell/.build/dapps/0x0587.../index.html
+  // Dapps installed from the registry and served by Parity; URL: http://127.0.0.1:8545/ff19...
+  const [hash] = window.location.pathname.match(/(0x)?[a-f0-9]{64}/i) || [];
+
+  // Dapps served in development mode on a dedicated port; URL: http://localhost:3001/?appId=dapp-name
+  const fromQuery = qs.parse(window.location.search).appId;
+
+  // Dapps built locally and served by Parity; URL: http://127.0.0.1:8545/dapp-name
+  const [, fromParity] = window.location.pathname.match(/^\/?([^/]+)\/?$/) || [];
 
-  let appId = match ? match[0] : query.appId;
+  return hash || fromQuery || fromParity;
```

Comment:
Could you make it such that if hash is found, then don't execute qs.parse, if fromQuery is found, don't execute the 3rd regex? To avoid unnecessary calculations.